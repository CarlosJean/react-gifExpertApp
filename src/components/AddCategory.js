import { useState } from "react";
import { PropTypes } from "prop-types";

export const AddCategory = ({setCategories}) => {

    const [category, setCategory] = useState('');

    const handleSubmit = (e)=>{
        e.preventDefault();        
        if (category.trim().length > 2) {
            setCategories( categories =>[category, ...categories]);
            setCategory('');
        }
    };

    const handleOnChange = (e) => {
        setCategory(e.target.value);         
    }

    return (
        <form onSubmit={handleSubmit}>
            <p>{category}</p>
            <input 
                type="text"
                placeholder="Ingrese algo"
                onChange={handleOnChange}
                value={category}
            />
            <button type="reset">Limpiar</button>
        </form>
    );
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}