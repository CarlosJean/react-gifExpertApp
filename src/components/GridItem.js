import PropTypes from "prop-types";

export const GridItem = ({title, url}) => {
    return (
        <div className="card animate__animated animate__fadeIn">
            <img 
                alt={title}
                src={url}
            />
            <p>{title}</p>
        </div>
    );
}

GridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}