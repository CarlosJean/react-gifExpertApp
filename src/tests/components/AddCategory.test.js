import {shallow} from 'enzyme';
import { AddCategory } from '../../components/AddCategory';
import '@testing-library/jest-dom'; //Para hacer mock de una funci√≥n.


describe('Pruebas al componente <AddCategory />', () => {

    const setCategories = jest.fn();
    let wrapper;

    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    });

    test('Debe mostrar el componente <AddCategory />', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe cambiar la caja de texto.', ()=>{
        const input = wrapper.find('input');
        const value = 'Hola Mundo';

        input.simulate('change', {target:{value}});
        expect(wrapper.find('p').text().trim()).toBe(value);

    });

    test('No debe realizar el submit', () => {
        wrapper.find('form').simulate('submit', {preventDefault(){}});

        expect(setCategories).not.toHaveBeenCalled();
    });
    
    test('Debe realizar el submit', () => {
        const value = 'Hola Mundo';
        
        const input = wrapper
        .find('input')
        .simulate('change', {target:{value}});

        wrapper
        .find('form')
        .simulate('submit', {preventDefault(){}});
        
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(input.prop('value')).toBe('');
    });
});