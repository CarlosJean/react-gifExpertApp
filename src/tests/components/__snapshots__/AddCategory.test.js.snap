// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <AddCategory /> Debe mostrar el componente <AddCategory /> 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Ingrese algo"
    type="text"
    value=""
  />
  <button
    type="reset"
  >
    Limpiar
  </button>
</form>
`;
